0000-stream/0000-stream.md

```diff
@@ -0,0 +1,405 @@
+| `0x13` | Stream Money Blocked |
+| `0x14` | Stream Data |
+| `0x15` | Stream Data Max |
+| `0x16` | Stream Data Blocked |
+
+#### 5.3.1. `ConnectionClose` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Error Code | UInt8 | Machine-readable [Error Code](#5-4-error-codes) indicating why the connection was closed. |
+| Error Message | Utf8String | Human-readable string intended to give more information helpful for debugging purposes. |
+
+If implementations allow half-open connections, an endpoint MAY continue sending packets after receiving a `ConnectionClose` frame. Otherwise, the endpoint MUST close the connection immediately.
+
+#### 5.3.2. `ConnectionNewAddress` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Source Address | ILP Address | New ILP address of the endpoint that sent the frame. |
+
+#### 5.3.3. `ConnectionMaxData` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Max Offset | VarUInt | The total number of bytes the endpoint is willing to receive on this connection. |
+
+Endpoints MUST NOT exceed the total number of bytes the other endpoint is willing to accept.
+
+#### 5.3.4. `ConnectionDataBlocked` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Max Offset | VarUInt | The total number of bytes the endpoint wants to send. |
+
+#### 5.3.5. `ConnectionMaxStreamId` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Max Stream ID | VarUInt | The maximum stream ID the endpoint is willing to accept. |
+
+#### 5.3.6. `ConnectionStreamIdBlocked` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Max Stream ID | VarUInt | The maximum stream ID the endpoint wishes to open. |
+
+#### 5.3.7. `StreamClose` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Stream ID | VarUInt | Identifier of the stream this frame refers to. |
+| Error Code | UInt8 | Machine-readable [Error Code](#5-4-error-codes) indicating why the stream was closed. |
+| Error Message | Utf8String | Human-readable string intended to give more information helpful for debugging purposes. |
+
+If implementations allow half-open streams, an endpoint MAY continue sending money or data for this stream after receiving a `StreamClose` frame. Otherwise, the endpoint MUST close the stream immediately.
+
+#### 5.3.8. `StreamMoney` Frame
+
+| Field | Type | Description |
+|---|---|---|
+| Stream ID | VarUInt | Identifier of the stream this frame refers to. |
+| Shares | VarUInt | Proportion of the ILP Prepare `amount` destined for the stream specified. |
+
+The amount of money that should go to each stream is calculated by dividing the number of shares for the given stream by the total number of shares in all of the `StreamMoney` frames in the packet.
+
+For example, if an ILP Prepare packet has an amount of 100 and three `StreamMoney` frames with 5, 15, and 30 shares for streams 2, 4, and 6, respectively, that would indicate that stream 2 should get 10 units, stream 4 gets 30 units, and stream 6 gets 60 units.
```

Comment:
Need to specify how to deal with rounding. i.e. If the amount that arrived in this case was 101 units and they are not divisible.

Simple solution may be to always round down and then allocate any remainder to the lowest numbered stream (or have a way to designate a default stream).
i.e. In this case stream 2 should get 11 units, stream 4 gets 30 units, and stream 6 gets 60 units.